lib/src/hydrated_bloc_storage.dart

```diff
@@ -67,4 +70,17 @@ class HydratedBlocStorage implements HydratedStorage {
     _instance = null;
     return await _file.exists() ? await _file.delete() : null;
   }
+
+  static Future<Directory> _getDocumentDir(Platform platform) async {
+    if (platform.isMacOS || platform.isLinux) {
```

Comment:
FYI, forcing this codepath will actually make your package behave worse for people using [a desktop path_provider implementation](https://github.com/google/flutter-desktop-embedding/tree/master/plugins/flutter_plugins/path_provider_fde) than if you left the original codepath.

Currently it's macOS only, but it would be easy for someone interested in supporting Windows or Linux to add an implementation to the FDE plugin.

Reply:
@stuartmorgan yeah I talked to @AppleEducate about reverting these changes and just trying to facilitate the updates in the lower level dependencies.

Reply:
Yes, the FDE implementations of plugins from flutter/plugins will migrate into flutter/plugins as each platform stabilizes. macOS implementations will start moving soon.

Even before that though, all someone needs to do is add the FDE plugin as a dependency in their project and anything using the main plugin will Just Work, due to the way method channels are implemented. So for macOS, all someone would need to do to use your package without the changes here would be to add path_provider_fde to their project. Once Windows and Linux implementations of path_provider_fde are in place, the same would be true there.

Reply:
Awesome, I've just tested it on MacOS and created https://github.com/felangel/hydrated_bloc/pull/24. Can you please take a look?